schema/field.go

```diff
@@ -35,6 +35,10 @@ type Field struct {
 // Validate validates the field.
 func (c Field) Validate(typeSet TypeSet) error {
 	// valid name
+	if c.Name == "" {
+		return fmt.Errorf("field name cannot be empty, might be missing the named key codec")
+	}
+
 	if !ValidateName(c.Name) {
```

Comment:
Does ValidateName not handle this case?

Reply:
Yes, it does handle it but I thought it was nice to have a more descriptive error for the empty case which would be the most common while implementing